### PR TITLE
Merge cardano-sl-0.5 to master

### PR DESCRIPTION
**Pull request by Martoon-00**
This includes changes made in #1316

### CODE REVIEW

File: src/Pos/Client/Txp/History.hs [hunk 1]
```diff
@@ -92,13 +92,16 @@ getSenders UnsafeTx {..} = do
 data TxHistoryEntry = THEntry
     { _thTxId        :: !TxId
     , _thTx          :: !Tx
-    , _thInputs      :: ![TxOut]
     , _thDifficulty  :: !(Maybe ChainDifficulty)
-    , _thInputAddrs  :: ![Address]  -- TODO: remove in favor of _thInputs
```
Comment (akegalj) [inline src/Pos/Client/Txp/History.hs]: is frontend UI using this field ? If yes, we should notify them how to get Addresses without that field @Martoon-00

Comment (Martoon-00) [inline src/Pos/Client/Txp/History.hs]: This is fully internal type. Client api operates with `CTx` instead, which remained unchanged

**Comment by gromakovsky** [comment]
@Martoon-00 there are conflicts, resolve them please. Also please request review more intrusively.

**Comment by Martoon-00** [comment]
❤️

**Pull request merged**